### PR TITLE
fix: config proxy-rewrite out of the plugin market in route module

### PR DESCRIPTION
**Pull request by liuxiran**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
fix: #1241 
___
### Bugfix
- Description

- How to fix?
image

add a new PanelSection called `Request Override` in Step 1 of create/edit route
___
### New feature or improvement
- Describe the details and related test reports.

___
### Please add the corresponding test cases if necessary.

- [x] add test case

___
### Backport patches
- Why need to backport?

- Source branch

- Related commits and pull requests

- Target branch

### CODE REVIEW

**Comment by juzhiyuan** [comment]
Wow! A big step!

**Comment by juzhiyuan** [comment]
cc @guoqqqi to take a review

**Comment by LiteSun** [comment]
CI failed.

File: web/cypress/integration/route/create-route-with-proxy-rewrite-plugin.spec.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+  it('should delete the route', function () {
+    cy.visit('/routes/list');
+    cy.get(this.domSelector.nameSelector).type(this.data.routeName);
+    cy.contains('Search').click();
+    cy.contains(this.data.routeName).siblings().contains('Delete').click();
+    cy.contains('button', 'Confirm').click();
+    cy.get(this.domSelector.notification).should('contain', this.data.deleteRouteSuccess);
+  });
+});
```
Comment (LiteSun) [inline web/cypress/integration/route/create-route-with-proxy-rewrite-plugin.spec.js:139]: Newline, please.😊

File: web/src/pages/Route/components/Step1/ProxyRewrite.tsx [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+            name={['proxyRewrite', 'regex_uri']}
+            initialValue={['', '']}
+          >
+            {(fields) =>
+              fields.map((field, index) => {
+                switch (index) {
+                  case 0:
+                    return (
+                      <Form.Item
+                        label={formatMessage({ id: 'page.route.form.itemLabel.regx' })}
```
Comment (LiteSun) [inline web/src/pages/Route/components/Step1/ProxyRewrite.tsx:71]: regex.

File: web/src/pages/Route/components/Step1/ProxyRewrite.tsx [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+            />
+          </Form.Item>
+        );
+      case HostRewriteType.KEEP:
+      default:
+        return null;
+    }
+  };
+
+  useEffect(() => {}, []);
```
Comment (LiteSun) [inline web/src/pages/Route/components/Step1/ProxyRewrite.tsx:157]: useless line? 🤔

Comment (liuxiran) [reply web/src/pages/Route/components/Step1/ProxyRewrite.tsx]: fixed

File: web/src/pages/Route/components/Step1/ProxyRewrite.tsx (hunk 2 above)
Comment (liuxiran) [reply web/src/pages/Route/components/Step1/ProxyRewrite.tsx]: fixed

**Comment by LiteSun** [comment]
CI failed.

File: web/cypress/integration/route/create-route-with-proxy-rewrite-plugin.spec.js (hunk 1 above)
Comment (liuxiran) [reply web/cypress/integration/route/create-route-with-proxy-rewrite-plugin.spec.js]: fixed

**Comment by liuxiran** [comment]
mark a unstable case: https://github.com/apache/apisix-dashboard/pull/1576/checks?check_run_id=2118588774
try to fix in : https://github.com/apache/apisix-dashboard/pull/1576/commits/92f2cd9b47334b82900f197b3963af8b754519bf

**Comment by liuxiran** [comment]
> CI failed.

errors fixed, PTLA again when you are free, thanks a lot.

**Review by LiteSun — APPROVED** [review]
LGTM

**Comment by imjoey** [comment]
@liuxiran the FE e2e test still seems broken. Shall we fix it beforehand?

File: web/cypress/integration/route/create-route-with-proxy-rewrite-plugin.spec.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    cy.get(this.domSelector.name).type(this.data.routeName);
+
+    // show requestOverride PanelSection
+    cy.contains(routeLocaleUS['page.route.panelSection.title.requestOverride']).should('be.visible');
+    // should show newPath after the uriRewriteType static clicked
+    cy.get(domSelector.staticUri).click();
+    cy.contains(routeLocaleUS['page.route.form.itemLabel.newPath']).should('be.visible');
+    cy.get(domSelector.newUri).should('be.visible').type(data.rewriteUri);
+    // should show regexp and template after uriRewriteType regexp clicked
+    cy.contains(routeLocaleUS['page.route.radio.regx']).click();
```
Comment (juzhiyuan) [inline web/cypress/integration/route/create-route-with-proxy-rewrite-plugin.spec.js:68]:
```suggestion
    cy.contains(routeLocaleUS['page.route.radio.regex']).click();
```

File: web/src/pages/Route/components/Step1/ProxyRewrite.tsx [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+import Form from 'antd/es/form';
+import { Button, Input, Radio, Row, Col } from 'antd';
+import { PlusOutlined, MinusCircleOutlined } from '@ant-design/icons';
+import { useIntl } from 'umi';
+import { PanelSection } from '@api7-dashboard/ui';
+
+import {
+  FORM_ITEM_LAYOUT,
+  FORM_ITEM_WITHOUT_LABEL,
+  ShcemeRewrite,
```
Comment (juzhiyuan) [inline web/src/pages/Route/components/Step1/ProxyRewrite.tsx:27]: Uppercase is used in Class IMO, we'd better use something like `A_B_C` in the future.

File: web/src/pages/Route/components/Step3/index.tsx [hunk 6]
```diff
@@ -32,16 +32,17 @@ type Props = {
   onChange: (data: { plugins: PluginComponent.Data; script: any, plugin_config_id?: string; }) => void;
   readonly?: boolean;
   isForceHttps: boolean;
+  isProxyEnable: boolean;
```
Comment (juzhiyuan) [inline web/src/pages/Route/components/Step3/index.tsx:35]: could it be optional and default to false?

**Review by juzhiyuan — COMMENTED** [review]
1 typo `regx` should be updated

**Comment by juzhiyuan** [comment]
also cc @guoqqqi to review.

**Comment by guoqqqi** [comment]
Please update the master branch code.

File: web/cypress/integration/route/create-route-with-proxy-rewrite-plugin.spec.js (hunk 4 above)
Comment (liuxiran) [reply web/cypress/integration/route/create-route-with-proxy-rewrite-plugin.spec.js]: done

File: web/src/pages/Route/components/Step1/ProxyRewrite.tsx (hunk 5 above)
Comment (liuxiran) [reply web/src/pages/Route/components/Step1/ProxyRewrite.tsx]: done

File: web/src/pages/Route/components/Step3/index.tsx (hunk 6 above)
Comment (liuxiran) [reply web/src/pages/Route/components/Step3/index.tsx]: yep, already fixed, thanks a lot

**Comment by liuxiran** [comment]
@imjoey @guoqqqi thanks for review, after update to the latest master branch, there are some new errors occurred in ci, let me try to fix it.

**Comment by liuxiran** [comment]
Screenshot_20210319_150946
Screenshot_20210319_152300

hi @juzhiyuan @LiteSun 

It seems that `netlify` errors caused by the build path of fe is `../output/html`, but the expected path is `web/dist`, we have to change the path to `output/html` in netlify configuration file. I need help that how can I change the configuration? looking forward to your reply, thanks.

**Comment by juzhiyuan** [comment]
@liuxiran just ignored it, Netlify still has some issues to resolve, I'm working on it.

**Comment by liuxiran** [comment]
> @liuxiran just ignored it, Netlify still has some issues to resolve, I'm working on it.

got 

> @liuxiran just ignored it, Netlify still has some issues to resolve, I'm working on it.

got it, thanks.

then since all the errors fixed, please take a look when you are free, thanks @juzhiyuan @LiteSun @imjoey @guoqqqi

**Review by imjoey — APPROVED** [review]
/lgtm

This is definitely a great step forward. Thanks you for this. @liuxiran

**Review by guoqqqi — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

File: web/src/pages/Route/Create.tsx [hunk 7]
```diff
@@ -93,6 +95,10 @@ const Page: React.FC<Props> = (props) => {
     }
   }, []);
 
+  const getProxyrewriteEnable =() => {
```
Comment (bzp2010) [inline web/src/pages/Route/Create.tsx:98]: This should be `getProxyRewriteEnable` ?

**Review by bzp2010 — COMMENTED** [review]
Is the comment a spelling problem?

**Review by LiteSun — APPROVED** [review]

File: web/src/pages/Route/Create.tsx (hunk 7 above)
Comment (juzhiyuan) [reply web/src/pages/Route/Create.tsx:98]: More like a spelling  issue 🤔 but not hurry

Comment (liuxiran) [reply web/src/pages/Route/Create.tsx]: fixed, thanks

**Review by bzp2010 — APPROVED** [review]

**Pull request merged**